nicegui/api_router.py

```diff
@@ -0,0 +1,43 @@
+from typing import Callable, Optional
+
+import fastapi
+
+from .page import page as ui_page
+
+
+class APIRouter(fastapi.APIRouter):
+
+    def page(self,
+             path: str, *,
+             title: Optional[str] = None,
+             viewport: Optional[str] = None,
+             favicon: Optional[str] = None,
+             dark: Optional[bool] = ...,
+             response_timeout: float = 3.0,
+             **kwargs,
+             ) -> Callable:
+        """Page
+
+        Creates a new page at the given route.
+        Each user will see a new instance of the page.
+        This means it is private to the user and not shared with others
+        (as it is done `when placing elements outside of a page decorator <https://nicegui.io/documentation#auto-index_page>`_).
+
+        :param path: route of the new page (path must start with '/')
+        :param title: optional page title
+        :param viewport: optional viewport meta tag content
+        :param favicon: optional relative filepath or absolute URL to a favicon (default: `None`, NiceGUI icon will be used)
+        :param dark: whether to use Quasar's dark mode (defaults to `dark` argument of `run` command)
+        :param response_timeout: maximum time for the decorated function to build the page (default: 3.0)
+        :param kwargs: additional keyword arguments passed to FastAPI's @app.get method
```

Comment:
Hi @rodja, 

I don't have much experience to draw from w/ FastAPI's `APIRouter`...

But, I would be curious to see or hear anecdotally of examples in which a NiceGUI user would pass through kwargs here.

This would help justify in my head that this addition ought to be considered an _extension_ of FastAPI's `APIRouter` (assuming a NiceGUI user still had practical use for its kwargs) or, if it should just be considered a a slightly _modified_ / _simplified_ version of the concept that allows for x, y, and z...

To reiterate more simply, would a NiceGUI user, in fact, ever _want_ to pass through `prefix`, `tags`, `dependencies`, or `responses` kwargs here?

Reply:
Thanks for the comment @sonnygeorge. Passing kwargs here (and the other params) are required to make it symetrical with `ui.page`. There, kwargs was introduced to fix #773.

In my opinion an extension to the existing APIRouter makes sense, because it allows you to mix pages with prefix with other "normal" FastAPI methods like `get` and `post`. With NiceGUI we want to make simple things simple and difficult things possible. In 98% of the cases you won't need kwargs, but if you need it you will be glad to have it.